src/main/java/org/opentripplanner/routing/algorithm/AStar.java

```diff
@@ -152,29 +153,40 @@ boolean iterate() {
 
         // get the lowest-weight state in the queue
         runState.u = runState.pq.extract_min();
-        
+
         // check that this state has not been dominated
         // and mark vertex as visited
         if (!runState.spt.visit(runState.u)) {
             // state has been dominated since it was added to the priority queue, so it is
             // not in any optimal path. drop it on the floor and try the next one.
             return false;
         }
-        
+
         if (traverseVisitor != null) {
             traverseVisitor.visitVertex(runState.u);
         }
-        
+
         runState.u_vertex = runState.u.getVertex();
 
         if (verbose)
             System.out.println("   vertex " + runState.u_vertex);
 
         runState.nVisited += 1;
-        
+
         Collection<Edge> edges = runState.options.arriveBy ? runState.u_vertex.getIncoming() : runState.u_vertex.getOutgoing();
         for (Edge edge : edges) {
 
+
+            if (this.runState.rctx.graph.carPresencePredictor != null &&
```

Comment:
It all should be inside `RentVehicleEdge#traverse`.

Reply:
This will also make this code testable, so please write tests ;)

Reply:
+2

Reply:
kiedyś ustalilismy, że z krawędzi nie da się dostać do grafu bez zmiany interfejsu

Reply:
oh, da się dostać do routingContext.
Miałem złe informacje na start

Reply:
At first I thought that we cannot reference Graph, but then I realized, that there is a reference in RoutingContext